Sources/Sentry/SentryCrashIntegration.m

```diff
@@ -49,30 +72,122 @@ - (BOOL)startCrashHandlerWithError:(NSError *_Nullable *_Nullable)error {
 }
 #pragma GCC diagnostic pop
 
-// TODO(fetzig) this was in client, used for testing only, not sure if we can still use this (for testing). maybe move it to hub or static-sdk?
+// TODO(fetzig) this was in client, used for testing only, not sure if we can
+// still use this (for testing). maybe move it to hub or static-sdk?
 - (void)reportUserException:(NSString *)name
                      reason:(NSString *)reason
                    language:(NSString *)language
                  lineOfCode:(NSString *)lineOfCode
                  stackTrace:(NSArray *)stackTrace
               logAllThreads:(BOOL)logAllThreads
            terminateProgram:(BOOL)terminateProgram {
+
     if (nil == installation) {
         [SentryLog logWithMessage:@"SentryCrash has not been initialized, call startCrashHandlerWithError" andLevel:kSentryLogLevelError];
         return;
     }
+
     [SentryCrash.sharedInstance reportUserException:name
-                                         reason:reason
-                                       language:language
-                                     lineOfCode:lineOfCode
-                                     stackTrace:stackTrace
-                                  logAllThreads:logAllThreads
-                               terminateProgram:terminateProgram];
+                                             reason:reason
+                                           language:language
+                                         lineOfCode:lineOfCode
+                                         stackTrace:stackTrace
+                                      logAllThreads:logAllThreads
+                                   terminateProgram:terminateProgram];
     [installation sendAllReports];
 }
 
 - (BOOL)crashedLastLaunch {
     return SentryCrash.sharedInstance.crashedLastLaunch;
 }
 
+- (void)addEventProcessor {
+    NSLog(@"addEventProcessor");
+    SentryEventProcessor eventProcessor = ^SentryEvent * _Nullable(SentryEvent * _Nonnull event) {
+
+        // skip early if integration (and therefore this event processor) isn't active on current client
+        if (NO == [SentrySDK.currentHub isIntegrationActiveInBoundClient:@"SentryCrashIntegration"]) {
```

Comment:
Can't we use `SentryCrashIntegration.class` here instead of a string?
This is not very safe when it comes to refactoring.

Reply:
forgot that as well. done